src/libraries/System.Private.CoreLib/src/System/Globalization/CultureData.Unix.cs

```diff
@@ -77,6 +71,62 @@ private bool InitCultureDataCore()
             return true;
         }
 
+        /// <summary>
+        /// Implementation of culture name validation.
+        /// </summary>
+        /// <remarks>
+        /// This is a fast approximate implementation based on BCP47 spec. It covers only parts of
+        /// the spec; such that, when it returns false, the input is definitely in incorrect format.
+        /// However, it returns true for some characters, which are not allowed by the spec, It also
+        /// returns true for some inputs where spec specifies the lengths of components, but we are
+        /// not validating components individually to limit its complexity at O(n) for now.
+        ///
+        /// Rules of implementation:
+        /// * Allow only letters, digits, - and '_' or \0 (NULL is for backward compatibility).
+        /// * Allow input length of zero (for invariant culture) or otherwise greater than 1 and less than or equal 85.
+        /// * Disallow input that starts or ends with '-' or '_'
+        /// * Disallow input that has any combination of consecutive '-' or '_'.
+        /// * Disallow input that has multiple '_'.
+        /// </remarks>
+        private static bool IsValidCultureName(string subject, out int firstIndexOfUnderscore)
+        {
+            firstIndexOfUnderscore = -1;
+
+            if (subject.Length == 0) return true; // Invariant Culture
+            if (subject.Length == 1 || subject.Length > 85) return false;
+
+            bool seenSeparator = false;
+            bool seenUnderscore = false;
+            for (int i = 0; i < subject.Length; ++i)
+            {
+                char c = subject[i];
+
+                if (c == '_' || c == '-')
+                {
+                    if (i == 0 || i == subject.Length - 1) return false;
+                    if (seenSeparator) return false;
```

Comment:
>seenSeparator [](start = 24, length = 13)

I don't think we need this variable. we can just check subject[i - 1] be '-' or '_'.

Reply:
OK. I kept it look-ahead, so we don't need to handle corner case for 0-th index.